src/Components/ThemesMenu/ThemesMenu.jsx

```diff
@@ -0,0 +1,123 @@
+import { useState } from "react";
+import Button from "../Button/Button";
+import "./ThemesMenu.css";
+
+export function ThemesMenu({
+  currentTheme,
+  onChange,
+  themes,
+  onAddTheme,
+  onEditTheme,
+  onDeleteTheme,
+}) {
+  const [newThemeName, setNewThemeName] = useState("");
+  const [isEditingTheme, setIsEditingTheme] = useState(false);
+  const [editThemeName, setEditThemeName] = useState(currentTheme.name);
+  const [isDelitingTheme, setIsDeletingTheme] = useState(false);
+
+  const handleThemeNameChange = (event) => {
+    setNewThemeName(event.target.value);
+  };
```

Comment:
if you also used the single state  you can update here the handler: ` setThemeState({ ...themeState, newThemeName: event.target.value 

- [ ] });`